apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/PublisherClient.cs

```diff
@@ -204,15 +239,23 @@ public static async Task<PublisherClient> CreateAsync(TopicName topicName, Clien
             var shutdowns = new Func<Task>[clientCount];
             for (int i = 0; i < clientCount; i++)
             {
-                var channelOptions = new[]
+                // Use a random arg to prevent sub-channel re-use in gRPC, so each channel uses its own connection.
+                var grpcChannelOptions = s_unlimitedSendReceiveChannelOptions
+                    .WithCustomOption("sub-channel-separator", Guid.NewGuid().ToString());
+
+                // First builder to handle any endpoint detection etc. We build a gRPC channel
+                // with this.
+                var builder = new PublisherServiceApiClientBuilder
```

Comment:
Just wondering if writing an `internal class ChannelBuilder : ClientBuilderBase<ChannelBase>` that `PublisherClient`, `PublisherServiceApiClientBuilder`, `SubscriberClient` and `SubscriberServiceApiClientBuilder` could rely on would simplify things.
The CallInvoker bits on this channel builder could be overwritten to throw, and this would be an internal class after all.
Not sure though.